### PR TITLE
Adding task commands to pulp-cli

### PR DESCRIPTION
**Pull request by daviddavis**
Will follow this up with another PR to break up core_cli into files like file_cli.

### CODE REVIEW

File: pulpcore/cli/core_cli.py [hunk 1]
```diff
@@ -8,3 +8,20 @@
...
+@main.group()
+@click.pass_context
+def task(ctx):
+    ctx.obj.href = "task_href"
+    ctx.obj.list_id = "tasks_list"
+
+
+@task.command()
+@click.option("--name")
+@click.option("--name__contains")
```
Comment (daviddavis) [inline pulpcore/cli/core_cli.py:22]: Perhaps this should be `--name-contains`? I'm torn between whether we want to be consistent with our API (to allow users to switch from the cli to the api) or to be consistent with CLIs in general.

Comment (mdellweg) [reply pulpcore/cli/core_cli.py:22]: I think, i'd prefer `--name-contains`. I guess you can specify the name of the variable this is attached to in the decorator.

File: pulpcore/cli/common.py [hunk 2]
```diff
@@ -20,9 +20,9 @@ def output_result(self, result):
         if self.format == "json":
             click.echo(json.dumps(result, cls=PulpJSONEncoder, indent=2))
 
-    def call(self, operation_id, *args, **kwargs):
+    def call(self, operation_id, background=False, *args, **kwargs):
         result = self.api.call(operation_id, *args, **kwargs)
-        if "task" in result:
+        if "task" in result and not background:
```
Comment (mdellweg) [inline pulpcore/cli/common.py:25]: In the case of background, the `click.echo(f"Started task {task_href}", err=True)` is maybe even more important.

File: pulpcore/cli/core/task.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+@task.command()
+@click.option("--href")
+@click.pass_context
+def cancel(ctx, href):
+    result = ctx.obj.call(
+        ctx.obj.cancel_id,
+        parameters={ctx.obj.href_key: href},
+        body={"state": "canceled"},
+    )
+    ctx.obj.output_result(result)
```
Comment (mdellweg) [inline pulpcore/cli/core/task.py:35]: If you add the wait for task here, the cli will wait for successful cancellation.

File: tests/scripts/test_task.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+. "$(dirname "$(realpath $0)")/config.sh"
+
+cleanup() {
+  pulp_cli file remote destroy --name "cli_test_file_remote"
+  pulp_cli file repository destroy --name "cli_test_file_repository"
+  pulp_cli orphans delete
+}
+trap cleanup EXIT
```
Comment (mdellweg) [inline tests/scripts/test_task.sh:10]: :100:

**Pull request closed without merging**